utils/utils.go

```diff
@@ -438,39 +438,30 @@ func DefaultJsonPostProcess( apiResponseMap map[generic_structs.ComparableApiReq
     parsedErrorStructure := make(map[string]interface{})
 
     for request, response := range apiResponseMap {
-
-        // Catch JSON slices that don't have a map at the root
-        var jsonSlice []interface{}
-        err := json.Unmarshal(response, &jsonSlice)
-        if err == nil {
-            LogWarn("DefaultJsonPostProcess", "JSON is a slice - building map.",
-                err)
```

Comment:
This is here for a reason and removing it will break other APIs.

JSON responses can come as a map {"key": "value"} or as a slice of maps [{"key": "value"}, {"key": "value"}].  While I wouldn't consider a slice "proper" JSON, it _is_ valid and at least a few of the plugins we already have depend on this.

This is a great reason why we need some automated testing/unit tests with different types of JSON payloads.

Reply:
In general, if you don't understand why a piece of code is there, it's better to ask then just remove it.  Especially with a codebase that isn't excessively unit tested like this.

It's good we have PRs/code review.